apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx

```diff
@@ -0,0 +1,198 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  title: 'On Delete Constraint Behavior',
+  description: 'Understand the types of foreign key constraint deletes',
+  footerHelpType: 'postgres',
+}
+
+There are 5 options for foreign key constraint deletes:
+
+1. **CASCADE:** When a row is deleted from the parent table, all related rows in the child table(s) are deleted as well.
+2. **RESTRICT:** When a row is deleted from the parent table, the delete operation is aborted if there are any related rows in the child table(s).
+3. **SET NULL:** When a row is deleted from the parent table, the values of the foreign key columns in the child table(s) are set to NULL.
+4. **SET DEFAULT:** When a row is deleted from the parent table, the values of the foreign key columns in the child table(s) are set to their default values.
+5. **NO ACTION:** This option is similar to RESTRICT, but it also has the option to be “deferred” to the end of a transaction. This means that other cascading deletes can run first, and then this delete constraint will only throw an error if there is referenced data remaining _at the end of the transaction_.
+
+These options can be specified when defining a foreign key constraint using the "ON DELETE" clause. For example, the following SQL statement creates a foreign key constraint with the `CASCADE` option:
+
+```sql
+alter table
+  child_table add constraint fk_parent foreign key (parent_id) references parent_table (id) on delete cascade;
+```
```

Comment:
is this prettier formatting it this way? Frustrating, I thought we had fixed it

Reply:
Yeah looks like it. Sorry didn't realize there was a history here! I imagine ideally this is wrapping sooner, right?

Reply:
Figured it out - 2 issues:
1. `prettier-plugin-sql-cst` wasn't working - I believe it was getting overridden by `prettier-plugin-sql` which is a sub dep of `@supabase/postgres-meta`. I fixed this by disabling that plugin and explicitly enabling this one.
2. Turns out `prettier-plugin-sql-cst` only understands SQLite, so any time it encounters Postgres-specific code that SQLite can't do (eg. dropping constraints), it fails to format. Not high priority issue as worst case we can just format manually - may not be solvable right now. Tracking here anyway:
  https://github.com/supabase/supabase/issues/14019